name-as/src/nma.rs

```diff
@@ -85,10 +87,62 @@ pub fn r_operation(mnemonic: &str) -> Result<R, &'static str> {
 /// Parses an I-type instruction mnemonic into an [I]
 pub fn i_operation(mnemonic: &str) -> Result<I, &'static str> {
     match mnemonic {
+        "ori" => Ok(I {
+            opcode: 0xd,
+            form: IForm::RtRsImm
+        }),
+        "lb" => Ok(I {
+            opcode: 0x20,
+            form: IForm::RtImmRs
+        }),
+        "lbu" => Ok(I {
+            opcode: 0x24,
+            form: IForm::RtImmRs
+        }),
+        "lh" => Ok(I {
+            opcode: 0x21,
+            form: IForm::RtImmRs
+        }),
+        "lhu" => Ok(I {
+            opcode: 0x25,
+            form: IForm::RtImmRs
+        }),
+        "lw" => Ok(I {
+            opcode: 0x23,
+            form: IForm::RtImmRs
+        }),
+        "ll" => Ok(I {
+            opcode: 0x30,
+            form: IForm::RtImmRs
+        }),
         "lui" => Ok(I {
             opcode: 0xf,
             form: IForm::RtImm,
         }),
+        "sb" => Ok(I {
+            opcode: 0x28,
+            form: IForm::RtImmRs
+        }),
+        "sh" => Ok(I {
+            opcode: 0x29,
+            form: IForm::RtImmRs
+        }),
+        "sw" => Ok(I {
+            opcode: 0x2b,
+            form: IForm::RtImmRs
+        }),
+        "sc" => Ok(I {
+            opcode: 0x38,
+            form: IForm::RtImmRs
+        }),
+        "beq" => Ok(I {
+            opcode: 0x4,
+            form: IForm::RsRtImm
```

Comment:
this needs to consider and resolve labels instead of just having immediate offsets. I'll approve this PR since you're probably already aware of this

Reply:
Yeah it just isn't a part of the parsed language yet, that'll be its own PR